server/src/main/java/org/elasticsearch/action/admin/cluster/storedscripts/PutStoredScriptRequest.java

```diff
@@ -166,4 +168,16 @@ public String toString() {
             (context != null ? ", context [" + context + "]" : "") +
             ", content [" + source + "]}";
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        if (source == null) {
```

Comment:
when is this if exercised? I think that you can make the blank constructor package protected and assume that once toXContent is called the source has already a value. that has the nice effect that we don't modify the state of the request when toXContent is called which is not ideal.